packages/language-server/tests/definitions.test.js

```diff
@@ -132,6 +132,22 @@ test('resolve markdown link references', async () => {
   ])
 })
 
+test('does not resolve shadow content', async () => {
```

Comment:
what do you mean by "shadow content"
Do you mean a JS context which has a shadow variable?

```
const a = 'root';

() => {
  const a = 'shadow';
}
```

or something else?

Reply:
The following MDX:

```mdx
<Hello />

export function Hello() {
  return 'hello'
}
```

is transformed to the following JSX:

```jsx
·········

export function Hello() {
  return 'hello'
}

export function MDXContent(props) {
  return <>
<Hello />

·························
················
·
</>
}
```

I call this the shadow content. I recall having seen this term somewhere, but I can’t find it anymore. I find it appropriate, but I’m open to other terminology. It’s purely internal.

This JSX is processed by TypeScript. The positions in the results are then mapped back to the original content.

In this case, TypeScript will report the `props` argument is unused. However, this is reported on a part of the shadow content that doesn’t map to the original MDX content. Previously this diagnostic was mapped to an incorrect position, leading to a false positive. It’s now stripped. The same goes for various API calls.

Reply:
Since this is just about internals, are you ok with me merging and releasing this? I’m pretty eager to release this, since it significantly increases the user experience.